src/Scenario/Merger.php

```diff
@@ -0,0 +1,232 @@
+        '@name',
+        '@type',
+    ];
+
+    /**
+     * @var array
+     */
+    private static $argumentRequiredArgs = [
+        '@name',
+        '@xsi:type'
+    ];
+
+    /**
+     * @var array
+     */
+    private static $itemRequiredArgs = [
+        '@name',
+        '@xsi:type'
+    ];
+
+    /**
+     * @param File $file
+     * @param XmlEncoder $encoder
+     * @param SystemList $systemList
+     * @param Resolver $resolver
+     */
+    public function __construct(File $file, XmlEncoder $encoder, SystemList $systemList, Resolver $resolver)
+    {
+        $this->file = $file;
+        $this->encoder = $encoder;
+        $this->systemList = $systemList;
+        $this->resolver = $resolver;
+    }
+
+    /**
+     * @param array $scenarios
+     * @return array
+     * @throws ValidationException
+     */
+    public function merge(array $scenarios): array
+    {
+        $data = [];
+
+        foreach ($scenarios as $scenario) {
+            $scenario = $this->collectScenario($scenario);
+
+            foreach ($scenario['step'] ?? [] as $step) {
+                if ($missedArgs = array_diff(self::$stepRequiredArgs, array_keys($step))) {
+                    throw new ValidationException(sprintf(
+                        'Arguments %s are missed from step',
+                        implode(', ', $missedArgs)
+                    ));
+                }
+
+                $data[$step['@name']] = array_replace_recursive(
+                    $data[$step['@name']] ?? [],
+                    $this->collectStep($step)
+                );
+            }
+        }
+
+        return $this->resolver->resolve($data);
+    }
+
+    /**
+     * @param array $step
+     * @return array
+     * @throws ValidationException
+     */
+    private function collectStep(array $step): array
+    {
+        $arguments = [];
+
+        foreach ($step['arguments'][0]['argument'] ?? [] as $argument) {
+            if ($missedArgs = array_diff(self::$argumentRequiredArgs, array_keys($argument))) {
+                throw new ValidationException(sprintf(
+                    'Arguments %s are missed from argument',
+                    implode(', ', $missedArgs)
+                ));
+            }
+
+            if ($argument['@xsi:type'] !== self::XSI_TYPE_ARRAY) {
+                throw new ValidationException(sprintf(
+                    'xsi:type %s not allowed in argument',
+                    $argument['@xsi:type']
+                ));
+            }
+
+            $arguments[] = [
+                'name' => $argument['@name'],
+                'xsi:type' => $argument['@xsi:type'],
+                'items' => $this->collectItems(
+                    $argument['item'] ?: []
+                )
+            ];
+        }
+
+        $stepData = [
+            'name' => $step['@name'],
+            'type' => $step['@type'],
+            'arguments' => $arguments
+        ];
+
+        if (isset($step['@remove'])) {
```

Comment:
where do you use this 'remove' element ?
And if step with this element is removed, why do you collect step arguments?